guides/source/action_cable_overview.md

```diff
@@ -0,0 +1,541 @@
+
+require 'action_cable/process/logging'
+
+run ActionCable.server
+```
+
+Then you start the server using a binstub in bin/cable ala:
+```
+#!/bin/bash
+bundle exec puma -p 28080 cable/config.ru
+```
+
+The above will start a cable server on port 28080. Remember to point your client-side setup against that using something like:
+`App.cable = Cable.createConsumer("ws://basecamp.dev:28080")`.
+
+### In app
+
+If you are using a threaded server like Puma or Thin, the current implementation of ActionCable can run side-along with your Rails application. For example, to listen for WebSocket requests on `/websocket`, match requests on that path:
+
+```ruby
+# config/routes.rb
+Example::Application.routes.draw do
+  match "/websocket", to: ActionCable.server, via: [:get, :post]
+end
+```
+
+You can use `App.cable = Cable.createConsumer("/websocket")` to connect to the cable server.
+
+For every instance of your server you create and for every worker your server spawns, you will also have a new instance of ActionCable, but the use of Redis keeps messages synced across connections.
+
+### Notes
+
+Beware that currently the cable server will _not_ auto-reload any changes in the framework. As we've discussed, long-running cable connections mean long-running objects. We don't yet have a way of reloading the classes of those objects in a safe manner. So when you change your channels, or the model your channels use, you must restart the cable server.
+
+The WebSocket server doesn't have access to the session, but it has access to the cookies. This can be used when you need to handle authentication. You can see one way of doing that with Devise in this [article](http://www.rubytutorial.io/actioncable-devise-authentication).
+
+## Dependencies
+
+Action Cable utilizes the pubsub feature of the configured storage adapter to route
+messages back and forth over the WebSocket cable connection. The default storage adapter
+is Redis, but Action Cable also provides an adapter for PostgreSQL. An alternate adapter
+can be used by adding the necessary adapter gem to the Gemfile.
+
+The Ruby side of things is built on top of [faye-websocket](https://github.com/faye/faye-websocket-ruby) and [concurrent-ruby](https://github.com/ruby-concurrency/concurrent-ruby).
+
+## Deployment
+
+Action Cable is powered by a combination of EventMachine and threads. The
+framework plumbing needed for connection handling is handled in the
+EventMachine loop, but the actual channel, user-specified, work is handled
+in a normal Ruby thread. This means you can use all your regular Rails models
+with no problem, as long as you haven't committed any thread-safety sins.
+
+But this also means that Action Cable needs to run in its own server process.
+So you'll have one set of server processes for your normal web work, and another
+set of server processes for the Action Cable. The former can be single-threaded,
+like Unicorn, but the latter must be multi-threaded, like Puma.
```

Comment:
This little paragraph is a little misleading. It gives the impression that Action Cable cannot run side-by-side with a Rails server in the same process, but it can. The emphasis should be that Action Cable itself needs a threaded server like Puma or Thin and will not work with a single-threaded server like Unicorn.